Build fails with go1.16 but works with 1.17, Go.mod points to go1.16

The commit 3841af9 updates helm to use go 1.17 on the CI but we have this on master:

https://github.com/helm/helm/blob/86a94f2f53b0519a3b22648a11f332c85cd8da1a/go.mod#L3

When building I encountered this error:

```
/data/go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
```

And found [here](https://github.com/kubernetes-sigs/controller-tools/issues/643#issuecomment-1004459362)  that updating to go1.17 fixes the issue.

I think, we need to update the go version in go.mod to 1.17.

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module helm.sh/helm/v3
 
-go 1.16
+go 1.17
 
 require (
 	github.com/BurntSushi/toml v0.4.1

